### ISSUE TITLE
R Package to make Shiny Electron apps

### ISSUE THREAD

**Issue by chasemc** [chirunconf/chirunconf19#12]
This is more of a from-scratch project if anyone is interested (even if during the unconf I'm not 🙃). A couple of weeks ago I quickly put some very-preliminary things here: https://github.com/chasemc/electricShine

I haven't had much time recently but I think it would be cool to have an R package that makes it easy for people to make electron-apps of shiny apps, like was done in the proof-of-concept here: https://github.com/ColumbusCollaboratory/electron-quick-start
Simply put- not everyone can afford the the cost of scaling in cloud-based deployments, especially data and computation intensive apps. 

Making electron apps would allow shiny-app-makers to create cross-platform apps that are easy for non-R-users to install locally. 

Previous efforts include the following:
https://github.com/wleepang/DesktopDeployR
https://github.com/ficonsulting/RInno
https://github.com/ColumbusCollaboratory/electron-quick-start

**Comment by jdblischak** [comment]
@chasemc This sounds like a great idea! I'd like to help if I have time (my main focus during the unconference will be mentoring new open source contributors). Have you tried RInno? I skimmed the docs, and it appears to be an R package that facilitates running a Shiny app with Electron. Is it lacking features you'd like to have?

**Comment by chasemc** [comment]
@dblischak  I've actually made some contributions to RInno...  

There's a few things that have made me think there needs to be a new package:
- RInno is focused on Inno and so is kind of, at its core, for creating Windows-only apps
- It can be complicated to use... and for my own uses I've had to customize the source code. 
- The electron part of RInno is new and still pretty much focused at Windows.
- The way RInno was written requires the shiny app be written as the two-file ui.r and server.r setup, and handles app package dependencies through RInno-  this is not ideal/not the way to create large apps (writing it as a package is).
- Lastly, a package that take a shiny app and creates an electron app could likely be an import within RInno. 

Also maybe relevant: I saw @revodavid is coming, even if it's just briefly on the side I'd love the chance to talk about the possibility of using Azure devops for shiny CI/CD, I've been messing around with it with R and electron but have to admit I'm pretty lost 😆.

**Comment by emilyriederer** [comment]
PLEASE do this! Also, @chasemc, I would love to pick you brain some on RInno in general. I've been thinking about trying to use it for some work apps (https://twitter.com/i/web/status/1087703980044681216) , but haven't actually dug in too much yet. The limitations  you list out here are actually super helpful.

**Comment by chasemc** [comment]
I've added a kanban board of initial things. Not comprehensive but hopefully is a good start
https://github.com/chasemc/electricShine/projects/1
